File: payroll-api/src/main/java/com/company/payroll/mapper/StaffAttendanceMapper.java
package com.company.payroll.mapper;

import com.company.payroll.model.StaffAttendance;
import org.apache.ibatis.annotations.Param;
import org.springframework.stereotype.Repository;

import java.time.LocalDate;
import java.util.List;

@Repository
public interface StaffAttendanceMapper {
    int deleteByPrimaryKey(Integer atId);
    int insert(StaffAttendance row);
    int insertSelective(StaffAttendance row);
    StaffAttendance selectByPrimaryKey(Integer atId);
    List<StaffAttendance> selectByStaffIdAndDate(@Param("staffId") Integer staffId,
                                                 @Param("startDate") LocalDate startDate,
                                                 @Param("endDate") LocalDate endDate);
    int updateCheckTime(StaffAttendance row);
    int updateByPrimaryKeySelective(StaffAttendance row);
    int updateByPrimaryKey(StaffAttendance row);
}

File: kotlin-payroll/src/main/kotlin/com/company/payroll/mapper/StaffAttendanceMapper.kt
package com.company.payroll.mapper

import com.company.payroll.model.StaffAttendance
import org.apache.ibatis.annotations.Param
import org.springframework.stereotype.Repository
import java.time.LocalDate

@Repository
interface StaffAttendanceMapper {
  fun deleteByPrimaryKey(atId: Int): Int

  fun insert(row: StaffAttendance): Int

  fun insertSelective(row: StaffAttendance): Int

  fun selectByPrimaryKey(atId: Int): StaffAttendance

  fun selectByStaffIdAndDate(@Param("staffId") staffId: Int,
                             @Param("startDate") startDate: LocalDate,
                             @Param("endDate") endDate: LocalDate): List<StaffAttendance>

  fun updateCheckTime(row: StaffAttendance): Int

  fun updateByPrimaryKeySelective(row: StaffAttendance): Int

  fun updateByPrimaryKey(row: StaffAttendance): Int
}